site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md

```diff
@@ -0,0 +1,135 @@
+---
+title: "Skip the Boilerplate: A Plugin Ready For Your Business Logic"
+image: /img/sonobuoy.svg
+excerpt: A new plugin helper library along with the new, upstream e2e-framework helps you jump straight into your testing your business logic. 
+author_name: John Schnake
+author_url: https://github.com/johnschnake
+categories: [kubernetes, sonobuoy, plugins]
+tags: ['Sonobuoy Team', 'John Schnake']
+date: 2021-11-09
+slug: plugin-starter
+---
+
+## Introduction
+
+When you decide to test your Kubernetes native code it can be a big chore. When I've been in that position, I do one of the following:
+ - start a new project
+   - Simple in concept but you're left with a ton of boilerplate code so that you can do simple things like
+ check if your app is running. You've got to deal with setting up client-go, you've got to figure out all the wiring
+ to set up builds, images, API clients, avoid test collision in the cluster, etc.
+ - vendor upstream k8s testing code
+   - This has the benefit of giving you lots of helper functions, but it is a huge dependency and bloats your tests. In addition,
+ even with all the new helpers, there is SO much code that it can be confusing about what to use and how to use it.
+
+Instead, we want to suggest you try our new [e2e-skeleton plugin][skeleton]. It is obviously just a starting point (not a plugin you would
```

Comment:
Change  ~`It is obviously`~ > `This plugin provides a starting point for creating and handling...`